[BUG] static "/usr/local/cuda" path string in cuda_lib cmake codegen

**🐛 Bug**
<!-- A clear and concise description of what the bug is. -->
Although the cmake file in nnfusion_rt folder can search CUDA path, the linking procedure still uses "/usr/local/cuda" path to link cuda libraries. It will result in linking error when users set custom CUDA paths.

**To Reproduce**
Steps to reproduce the behavior:
1. set custom CUDA path (e.g., /usr/local/cuda-10.2)
2. remove "/usr/local/cuda" or redirect "/usr/local/cuda" to another CUDA path (e.g., /usr/local/cuda -> /usr/local/cuda-9.0)
3. compile model and build nnfusion_rt
<!-- If you have a code sample, error messages, stack traces, please provide it here as well -->

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
```bast
cmake .
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found CUDA: /usr/local/cuda-10.2 (found version "10.2")
-- Configuring done
-- Generating done
```

```bash
make -j
Scanning dependencies of target nnfusion_naive_rt
[ 96%] Linking CXX static library libnnfusion_naive_rt.a
[ 96%] Built target nnfusion_naive_rt
Scanning dependencies of target main_test
[ 98%] Building CXX object CMakeFiles/main_test.dir/main_test.cpp.o
[100%] Linking CXX executable main_test
/usr/bin/ld: cannot find -lcudnn
collect2: error: ld returned 1 exit status
CMakeFiles/main_test.dir/build.make:110: recipe for target 'main_test' failed
make[2]: *** [main_test] Error 1
CMakeFiles/Makefile2:96: recipe for target 'CMakeFiles/main_test.dir/all' failed
make[1]: *** [CMakeFiles/main_test.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make: *** [all] Error 2
```

**Additional context**
<!-- Add any other context about the problem here. -->
After setting soft link "/usr/local/cuda -> /usr/local/cuda-10.2", it works well.

--- a/src/nnfusion/engine/pass/codegen/codegen_langunit.cpp
+++ b/src/nnfusion/engine/pass/codegen/codegen_langunit.cpp
@@ -64,7 +64,7 @@ target_link_libraries(${TARGET_NAME}
 
 LU_DEFINE(nnfusion::codegen::cmake::cuda_lib,
           R"(
-link_directories(/usr/local/cuda/lib64)
+link_directories(${CUDA_TOOLKIT_ROOT_DIR}/lib64)
 
 find_path(CUDNN_INCLUDE_DIR cudnn.h
     HINTS ${CUDA_TOOLKIT_ROOT_DIR}
@@ -76,8 +76,8 @@ find_library(CUDNN_LIBRARY cudnn
     HINTS ${CUDA_TOOLKIT_ROOT_DIR}
     PATH_SUFFIXES lib lib64 cuda/lib cuda/lib64 lib/x64)
 
-find_library(CUDA_cuda_LIBRARY cuda /usr/local/cuda/lib64/stubs)
-find_library(CUDA_cudart_LIBRARY libcudart.so /usr/local/cuda/lib64)
+find_library(CUDA_cuda_LIBRARY cuda ${CUDA_TOOLKIT_ROOT_DIR}/lib64/stubs)
+find_library(CUDA_cudart_LIBRARY libcudart.so ${CUDA_TOOLKIT_ROOT_DIR}/lib64)
 
 target_link_libraries(${TARGET_NAME}
     ${CUDA_cuda_LIBRARY}

--- a/src/nnfusion/engine/pass/codegen/cuda_codegen_pass.cpp
+++ b/src/nnfusion/engine/pass/codegen/cuda_codegen_pass.cpp
@@ -1049,6 +1049,7 @@ cmake_minimum_required(VERSION 3.5)
 
 SET(SRC "nnfusion_rt.cu" CACHE STRING "codegen source file")
 SET(TARGET_NAME "nnfusion_naive_rt" CACHE STRING "codegen target name")
+SET(CUDA_ARCH "-gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75" CACHE STRING "target architecture")
 
 if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
@@ -1059,7 +1060,7 @@ set(CMAKE_CXX_FLAGS_DEBUG "-g")
 set(CMAKE_CXX_FLAGS_RELEASE "-O2")
 
 find_package(CUDA)
-set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75")
+set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} ${CUDA_ARCH}")
 set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -O2")
 set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -cudart shared")
 )";

